lib/broccoli/amd-excluder.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const Funnel = require('broccoli-funnel');
+const walkSync = require('walk-sync');
+
+class AmdExcluder extends Funnel {
+  constructor(inputNode, options) {
+    super(inputNode, {
+      exclude: [],
+      annotation: options.annotation,
+    });
+
+    this.options = options;
+  }
+
+  build() {
+    let inputPath = this.inputPaths[0];
+
+    let files = walkSync(inputPath, {
+      directories: false,
+      globs: [`**/*.js`],
+    });
+
+    for (let file of files) {
+      let inputFilePath = path.join(inputPath, file);
+      let source = fs.readFileSync(inputFilePath, 'utf8');
+
+      // ember-data and others compile their own source
+      if (source.indexOf('define(') === 0) {
```

Comment:
Should be fine, but I'm not 100 about potential edgecases. It would likely not be compatible JS anyway so this should be good enough to figure out if something is AMD.

On the other hand, what if the module is exported for AMD but doesn't `define` immediately.

Reply:
There are a lot of potential edge cases, but in reality we only have to worry about how babel formats the AMD code, which this should cover completely?